### PR TITLE
[receiver/sqlserver] Enable more perf counter metrics

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

These metrics already exist, and are all enabled by default, but were previously not being recorded when directly connecting to the SQL Server instance. This means these metrics will now be enabled by default across all systems and configurations.

Metrics:
```
sqlserver.batch.request.rate
sqlserver.batch.sql_compilation.rate
sqlserver.batch.sql_recompilation.rate
sqlserver.page.buffer_cache.hit_ratio
sqlserver.user.connection.count
```

**Testing:** <Describe what testing was performed and which tests were added.>
Updated tests to account for given metrics

**Documentation**
Updated documentation to show these metrics are now available on more systems than just Windows.

### CODE REVIEW

File: .chloggen/sqlserver_pc_metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: sqlserverreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Enable more perf counter metrics when directly connecting to SQL Server
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (crobert-1) [inline .chloggen/sqlserver_pc_metrics.yaml:13]:
```suggestion
issues: [33420]
```

File: .chloggen/sqlserver_pc_metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
Comment (crobert-1) [inline .chloggen/sqlserver_pc_metrics.yaml:4]: I'm on the fence as to whether this should be considered an `enhancement` or `breaking` change. These metrics are all shown as being enabled by default, but the reality is they aren't received unless running on Windows.

This change does increase the number of metrics being received when directly connecting to the SQL server instance in reality, so let me know if we should mark this as `breaking` instead.

**Comment by crobert-1** [comment]
_Both CI/CD failures are unrelated to this change. I've added frequencies on relevant issues._

File: .chloggen/sqlserver_pc_metrics.yaml (hunk 2 above)
Comment (djaglowski) [reply .chloggen/sqlserver_pc_metrics.yaml:4]: Seems reasonable to call it an enhancement

**Review by djaglowski — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**